solver/src/driver.rs

```diff
@@ -94,30 +90,44 @@ impl Driver {
         let estimated_prices = self
             .price_estimator
             .estimate_prices(tokens.as_slice(), denominator_token)
-            .await
-            .context("failed estimating prices")?;
+            .await;
 
-        Ok(tokens.into_iter().zip(estimated_prices).collect())
+        tokens
+            .into_iter()
+            .zip(estimated_prices)
+            .filter_map(|(token, price)| match price {
+                Ok(price) => Some((token, price)),
+                Err(err) => {
+                    tracing::warn!("failed to estimate price for token {}: {:?}", token, err);
+                    None
+                }
+            })
+            .collect()
     }
 
     pub async fn single_run(&mut self) -> Result<()> {
         tracing::debug!("starting single run");
-        let limit_orders = self
+        let mut limit_orders = self
             .orderbook
             .get_liquidity()
             .await
             .context("failed to get orderbook")?;
         tracing::debug!("got {} orders", limit_orders.len());
 
+        let estimated_prices = self.collect_estimated_prices(&limit_orders).await;
```

Comment:
nanonit: Maybe log number of orders after the pruning?

Reply:
added